fastlane_core/lib/fastlane_core/keychain_importer.rb

```diff
@@ -32,7 +33,38 @@ def self.import_file(path, keychain_path, keychain_password: "", certificate_pas
       end
     end
 
-    def self.set_partition_list(path, keychain_path, keychain_password: "", output: FastlaneCore::Globals.verbose?)
+    def self.set_partition_list(path, keychain_path, keychain_password: nil, output: FastlaneCore::Globals.verbose?)
+      keychain_name = File.basename(keychain_path, ".*")
+      server = server_name(keychain_name)
+
+      # https://github.com/fastlane/fastlane/issues/14196
```

Comment:
Could this lookup of the keychain_password be moved into a function and called where `set_partition_list` gets called? i.e. decouple the lookup from this method (since it already accepts keychain password as a value)

Reply:
Yup yup, can do!